share/shwap/p2p/bitswap/block_store.go

```diff
@@ -0,0 +1,113 @@
+package bitswap
+
+import (
+	"context"
+	"fmt"
+
+	blocks "github.com/ipfs/go-block-format"
+	"github.com/ipfs/go-cid"
+
+	eds "github.com/celestiaorg/celestia-node/share/new_eds"
+	bitswappb "github.com/celestiaorg/celestia-node/share/shwap/p2p/bitswap/pb"
+)
+
+// Accessors abstracts storage system that indexes and manages multiple eds.Accessors by network height.
+type Accessors interface {
+	// Get returns an Accessor by its height.
+	Get(ctx context.Context, height uint64) (eds.Accessor, error)
+}
+
+// Blockstore implements generalized Bitswap compatible storage over Shwap containers
+// that operates with Block and accesses data through Accessors.
+type Blockstore struct {
+	Accessors
```

Comment:
Is there particular reason to have it as embedding?

Reply:
Not really